lib/trino-parquet/src/main/java/io/trino/parquet/reader/SimpleSliceInputStream.java

```diff
@@ -88,4 +96,21 @@ public int getByteArrayOffset()
     {
         return offset + slice.byteArrayOffset();
     }
+
+    public void ensureBytesAvailable(int bytes)
+    {
+        checkPositionIndexes(offset, offset + bytes, slice.length());
+    }
+
+    /**
+     * Use this method only if there is a significant performance boost validated in benchmarks.
```

Comment:
What's disadvantage of using it always?

Reply:
The only disadvantage is that it's "unsafe", so if we make an out-of-bounds access without doing bounds check first then we get a crash.